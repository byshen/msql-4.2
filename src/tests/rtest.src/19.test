#
# Test the sub-integer types
#


create table table_int64 (v1 int64)\p\g
create table table_uint64 (v1 uint64)\p\g

insert into table_uint64 values (2928512910929999932)\p\g
insert into table_uint64 values (2928512910929999930)\p\g
insert into table_uint64 values (2928512910939999932)\p\g
insert into table_uint64 values (29285129109299932)\p\g
insert into table_uint64 values (17916492470816866292)\p\g


select * from table_uint64 where v1 = 2928512910929999930\p\g
select * from table_uint64 where v1 > 2928512910929999930\p\g
select * from table_uint64 where v1 < 999999999999999999\p\g
select * from table_uint64 where v1 > 0\p\g
select * from table_uint64 where v1 between 2928512910929999930 
	and 2928512910929999999 \p\g

create index idx1 on table_uint64 (v1)\p\g

select * from table_uint64 where v1 = 2928512910929999930\p\g
select * from table_uint64 where v1 > 2928512910929999930\p\g
select * from table_uint64 where v1 < 999999999999999999\p\g
select * from table_uint64 where v1 > 0\p\g
select * from table_uint64 where v1 between 2928512910929999930 
	and 2928512910929999999 \p\g

select * from table_uint64 order by v1\p\g

insert into table_int64 values (1)\p\g
insert into table_int64 values (-530251602892685324)\p\g
insert into table_int64 values (999999999999999)\p\g

select * from table_int64 order by v1\p\g
