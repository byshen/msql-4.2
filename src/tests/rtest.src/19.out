
Welcome to the miniSQL monitor.  Type \h for help.


mSQL >     ->     -> 
Query buffer
------------


create table table_int64 (v1 int64)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
create table table_uint64 (v1 uint64)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
insert into table_uint64 values (2928512910929999932)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into table_uint64 values (2928512910929999930)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into table_uint64 values (2928512910939999932)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into table_uint64 values (29285129109299932)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into table_uint64 values (17916492470816866292)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     ->     -> 
Query buffer
------------
select * from table_uint64 where v1 = 2928512910929999930
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 2928512910929999930|
 +------------------+



mSQL > 
Query buffer
------------
select * from table_uint64 where v1 > 2928512910929999930
[continue]
    -> 
Query OK.  3 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 2928512910929999932|
 | 2928512910939999932|
 | 17916492470816866292|
 +------------------+



mSQL > 
Query buffer
------------
select * from table_uint64 where v1 < 999999999999999999
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 29285129109299932|
 +------------------+



mSQL > 
Query buffer
------------
select * from table_uint64 where v1 > 0
[continue]
    -> 
Query OK.  5 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 2928512910929999932|
 | 2928512910929999930|
 | 2928512910939999932|
 | 29285129109299932|
 | 17916492470816866292|
 +------------------+



mSQL >     -> 
Query buffer
------------
select * from table_uint64 where v1 between 2928512910929999930 
	and 2928512910929999999 
[continue]
    -> 
Query OK.  2 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 2928512910929999932|
 | 2928512910929999930|
 +------------------+



mSQL >     -> 
Query buffer
------------
create index idx1 on table_uint64 (v1)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
select * from table_uint64 where v1 = 2928512910929999930
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 2928512910929999930|
 +------------------+



mSQL > 
Query buffer
------------
select * from table_uint64 where v1 > 2928512910929999930
[continue]
    -> 
Query OK.  3 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 2928512910929999932|
 | 2928512910939999932|
 | 17916492470816866292|
 +------------------+



mSQL > 
Query buffer
------------
select * from table_uint64 where v1 < 999999999999999999
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 29285129109299932|
 +------------------+



mSQL > 
Query buffer
------------
select * from table_uint64 where v1 > 0
[continue]
    -> 
Query OK.  5 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 29285129109299932|
 | 2928512910929999930|
 | 2928512910929999932|
 | 2928512910939999932|
 | 17916492470816866292|
 +------------------+



mSQL >     -> 
Query buffer
------------
select * from table_uint64 where v1 between 2928512910929999930 
	and 2928512910929999999 
[continue]
    -> 
Query OK.  2 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 2928512910929999930|
 | 2928512910929999932|
 +------------------+



mSQL >     -> 
Query buffer
------------
select * from table_uint64 order by v1
[continue]
    -> 
Query OK.  5 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | 29285129109299932|
 | 2928512910929999930|
 | 2928512910929999932|
 | 2928512910939999932|
 | 17916492470816866292|
 +------------------+



mSQL >     -> 
Query buffer
------------
insert into table_int64 values (1)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into table_int64 values (-530251602892685324)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
insert into table_int64 values (999999999999999)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
select * from table_int64 order by v1
[continue]
    -> 
Query OK.  3 row(s) modified or retrieved.

 +------------------+
 | v1               |
 +------------------+
 | -530251602892685324|
 | 1                |
 | 999999999999999  |
 +------------------+



mSQL > 
Bye!

